assets/javascripts/discourse/components/chat-message-collapser.js

```diff
@@ -1,78 +1,125 @@
 import Component from "@ember/component";
-import { action, computed } from "@ember/object";
+import { computed } from "@ember/object";
+import { htmlSafe } from "@ember/template";
 import escape from "discourse-common/lib/escape";
 import domFromString from "discourse-common/lib/dom-from-string";
 import I18n from "I18n";
 
 export default Component.extend({
+  tagName: "",
+
   collapsed: false,
   uploads: null,
   cooked: null,
-  message: null,
 
-  @computed("uploads")
-  get imageFilename() {
-    if (this.uploads) {
-      if (this.uploads.length === 1) {
-        return this.uploads[0].original_filename;
+  @computed("cooked")
+  get youtubeCooked() {
+    const elements = Array.prototype.slice.call(domFromString(this.cooked));
+
+    return elements.reduce((acc, e) => {
+      if (youtubePredicate(e)) {
+        const id = e.dataset.youtubeId;
+        const link = `https://www.youtube.com/watch?v=${escape(id)}`;
+        const title = e.dataset.youtubeTitle;
```

Comment:
I think I'll escape it in case. I shouldn't make the assumption that everything is coming from https://github.com/discourse/discourse/blob/main/plugins/lazy-yt/assets/javascripts/lib/lazyYT.js